docs/no_peeking/mappings/family.py

```diff
@@ -0,0 +1,110 @@
+"""
+This module maps Kids First family to Phenopackets Family (derived from FHIR Group).
+Please visit https://aehrc.github.io/fhir-phenopackets-ig/StructureDefinition-Family.html
+    for the detailed structure definition.
+"""
+from kf_lib_data_ingest.common import constants
+from kf_lib_data_ingest.common.concept_schema import CONCEPT
+
+
+def family_type(x):
+    """
+    http://hl7.org/fhir/R4/valueset-group-type.html
+    """
+    if x in {constants.SPECIES.DOG}:
+        return 'animal'
+    elif x == constants.SPECIES.HUMAN:
+        return 'person'
+    else:
+        raise Exception('Unknown Family type')
+
+def family_member_type(x):
+    """
+    https://www.hl7.org/fhir/v3/FamilyMember/vs.html
+    """
+    if x == constants.COMMON.TRUE:
+        return {
+            'system': 'http://terminology.hl7.org/CodeSystem/v3-RoleCode',
+            'code': 'CHILD',
+            'display': 'child'
+        }
+    elif x == constants.RELATIONSHIP.FATHER:
+        return {
+            'system': 'http://terminology.hl7.org/CodeSystem/v3-RoleCode',
+            'code': 'FTH',
+            'display': 'father'
+        }
+    elif x == constants.RELATIONSHIP.MOTHER:
+        return {
+            'system': 'http://terminology.hl7.org/CodeSystem/v3-RoleCode',
+            'code': 'MTH',
+            'display': 'mother'
+        }
+    else:
+        raise Exception('Unknown Family family-member-type')
+
+def family_member_phenopacket(x): pass
+
+
+family = {
+    'resourceType': 'Group',
+    'id': CONCEPT.FAMILY.TARGET_SERVICE_ID,
+    'meta': {
+        'profile': ['http://ga4gh.fhir.phenopackets/StructureDefinition/Family']
+    },
+    'identifier': [
+        {
+            'system': 'https://kf-api-dataservice.kidsfirstdrc.org/families',
+            'value': CONCEPT.FAMILY.TARGET_SERVICE_ID
+        },
+        {
+            'value': CONCEPT.FAMILY.ID
+        }
+    ],
+    'extension': [
+        {
+            'text': 'PedigreeNodeReference',
+            'url': 'http://ga4gh.org/fhir/phenopackets/StructureDefinition/PedigreeNodeReference',
+            'valueReference': None
+        }
+    ],
+    'type': family_type(CONCEPT.PARTICIPANT.SPECIES or constants.SPECIES.HUMAN),
+    'actual': constants.COMMON.TRUE, # defaults to True
+    'member': [
```

Comment:
The "member" is missing the "family-member-phenopacket" extension shown here:

https://aehrc.github.io/fhir-phenopackets-ig/StructureDefinition-Family-definitions.html#Group.member.extension:family-member-phenopacket

Also, the new IG doesn't show the ValueSet/CodeSystme  that is used in the "family-member-type" extension. I think it was shown in the older IG format.